### PR TITLE
Add openapi validation method.

### LINKED ISSUE #2: Add validation for CRDs
We should add validation for CRDs. The Kubernetes libraries provide this functionality, but I held off on this due to import-dependency-issues.

### PR DESCRIPTION
**Pull request by Kashomon**
This method uses OpenAPI schemas to validate arbitrary JSON structures.
I moved this to an openapi directory from validate because this is also
where I will put defaulting logic. The openapi library takes the result
of a validate call and allows you to use it to other transformations
(e.g., defaulting).

Addresses #141 
Fixes #2

### CODE REVIEW

File: pkg/options/openapi/validate.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
+// Copyright 2018 Google LLC
```
Comment (johnbelamaric) [inline pkg/options/openapi/validate.go]: 2019

File: pkg/options/openapi/validate_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
+// Copyright 2018 Google LLC
```
Comment (johnbelamaric) [inline pkg/options/openapi/validate_test.go]: 2019

File: pkg/options/openapi/LICENSE [hunk 3]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (johnbelamaric) [inline pkg/options/openapi/LICENSE]: Why do we need this here? We already have it at the root level.

Comment (Kashomon) [reply pkg/options/openapi/LICENSE]: Well, I copied this file wholesale from K8S, so I felt like I needed to copy the license. WDYT?

**Comment by johnbelamaric** [comment]
No, I don't think so. If we had a different license, then possibly. But we
already have the same license. You have preserved the copyright notices,
that is sufficient.

File: pkg/options/openapi/LICENSE (hunk 3 above)
Comment (Kashomon) [reply pkg/options/openapi/LICENSE]: Removed.

File: pkg/options/openapi/validate.go (hunk 1 above)
Comment (Kashomon) [reply pkg/options/openapi/validate.go]: Done

File: pkg/options/openapi/validate_test.go (hunk 2 above)
Comment (Kashomon) [reply pkg/options/openapi/validate_test.go]: done

**Comment by Kashomon** [comment]
PTAL, fixed the copyright/license issues

**Pull request merged**